completions.go

```diff
@@ -117,22 +117,31 @@ type CompletionOptions struct {
 	HiddenDefaultCmd bool
 }
 
+// CompletionFunc is a function that provides completion results.
+type CompletionFunc func(cmd *Command, args []string, toComplete string) ([]string, ShellCompDirective)
+
 // NoFileCompletions can be used to disable file completion for commands that should
 // not trigger file completions.
+//
+// This method satisfies [CompletionFunc] and can be used with [Command.RegisterFlagCompletionFunc]
 func NoFileCompletions(cmd *Command, args []string, toComplete string) ([]string, ShellCompDirective) {
 	return nil, ShellCompDirectiveNoFileComp
 }
 
 // FixedCompletions can be used to create a completion function which always
 // returns the same results.
-func FixedCompletions(choices []string, directive ShellCompDirective) func(cmd *Command, args []string, toComplete string) ([]string, ShellCompDirective) {
+//
+// This method returns a function that satisfies [CompletionFunc] and can be used with [Command.RegisterFlagCompletionFunc]
+func FixedCompletions(choices []string, directive ShellCompDirective) CompletionFunc {
 	return func(cmd *Command, args []string, toComplete string) ([]string, ShellCompDirective) {
 		return choices, directive
 	}
 }
 
 // RegisterFlagCompletionFunc should be called to register a function to provide completion for a flag.
-func (c *Command) RegisterFlagCompletionFunc(flagName string, f func(cmd *Command, args []string, toComplete string) ([]string, ShellCompDirective)) error {
+//
+// You can use [FixedCompletions] or [NoFileCompletions] as completion functions.
```

Comment:
This comment gives the impression those two functions are the only one that can be used.
How about:
```
// You can use pre-defined completion functions such as [FixedCompletions] or [NoFileCompletions],
// or you can define your own.
```